server/queue/queue-memory/src/main/java/org/apache/james/queue/memory/MemoryMailQueueFactory.java

```diff
@@ -98,14 +100,26 @@ public String getName() {
 
         @Override
         public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueException {
-            ZonedDateTime nextDelivery = ZonedDateTime.now().plus(delay, Temporals.chronoUnit(unit));
+            ZonedDateTime nextDelivery = calculateNextDelivery(delay, unit);
             try {
                 mailItems.put(new MemoryMailQueueItem(cloneMail(mail), this, nextDelivery));
             } catch (MessagingException e) {
                 throw new MailQueueException("Error while copying mail " + mail.getName(), e);
             }
         }
 
+        private ZonedDateTime calculateNextDelivery(long delay, TimeUnit unit) {
+            if (delay > 0) {
+                try {
+                    ZonedDateTime.now().plus(delay, Temporals.chronoUnit(unit));
+                } catch (ArithmeticException e) {
+                    return Instant.ofEpochMilli(Long.MAX_VALUE).atZone(ZoneId.systemDefault());
```

Comment:
I prefer not using system timezones, maybe you can use UTC instead.

Reply:
fixup is available at `trantienduchn/JAMES-2309`